src/redux/modules/events.js

```diff
@@ -29,17 +29,22 @@ export const eventsRequestError = (error) => ({
 });
 
 // potentialIds (payload: { deviceid: String, data: String }) => String[]
-const potentialIds = ({ deviceid = '', data = '' }) => [
-  data.toLowerCase(),
-  data.toUpperCase(),
-  data.toLowerCase().replace(/^3000/, ''),
-  data.toUpperCase().replace(/^3000/, ''),
-  deviceid.toLowerCase(),
-  deviceid.toUpperCase()
-];
+export const potentialIds = ({ deviceid, data }) => {
+  // these now come through as `null` sometimes :S
+  deviceid = typeof deviceid === 'string' ? deviceid : '' + deviceid;
```

Comment:
The `"null"` and `"undefined"` strings won't match anything relevant in the details map, so this is fine for now. You're right though. I was lazy before when I was just trying to keep the arrow function simple, but now that I have ternaries in there with a full function body, I may as well do something smarter. Give me a second. :)